### ISSUE TITLE
REF-96: Code should follow the MVVM architecture

### ISSUE THREAD

**Issue by refatK** [Landry333/Big-Owl#96]
**Refactor Type(s):**

- [Encapsulate Record](https://refactoring.com/catalog/encapsulateRecord.html)
  - Repositories map database data onto Model classses, this lets us use the data directly like a java object rather than needing to use strings to refrence the data from the database

- [Separate Query from Modifier](https://refactoring.com/catalog/separateQueryFromModifier.html), [Move Function](https://refactoring.com/catalog/moveFunction.html) and [Replace Inline Code with Function Call](https://refactoring.com/catalog/replaceInlineCodeWithFunctionCall.html)
  - Query logic is separated to new class (Repository classes). Reduces coupling of logic. No more inline logic in UI classes increases readability of code

**Description of Epic**:
At the moment, the majority of our code uses the database directly from UI classes (Activities, Fragments, etc.). UI classes should only be concerned with UI logic, and not contain database query logic. For the sake of code readability and ease of testing, the code in the UI classes should be refactored such that **database logic** is moved to Repository classes instead. Moreover, ViewModels should be implemented to extract **business logic** from the UI. This is necessary for many parts of the UI code, hence why this is an epic.

In summary:
- Models = data mapping (Organization of the data)
- Repository = database logic (queries, etc.)
- ViewModels = business logic (Sign up, edit, adding users, remove users, etc.)
- Activities/Fragment/Adapter = UI logic (Visualization of the data, behaviour of the UI, etc.)

**Architecture link**: https://github.com/Landry333/Big-Owl/wiki/Architecture

**Comment by ShawnGuerra-Bautista** [comment]
This issue should be closed. All issues within the Epic were already done a while ago.
